### DISCUSSION TITLE
Output full document where a given line is found

### DISCUSSION THREAD

**Discussion by chipzoller** [mikefarah/yq#1853] [General]
Hi there, been using `yq` for a while and it's a tremendously useful tool. One thing I've been trying to do is a simple way to say, "give me the full YAML document found within this file (which contains multiple) where a given line is found." For example, you perform a `helm template` and get back a ton of resources, but I want to get only one of those but see the entire YAML for that resource. I'm trying to avoid using `grep -A` and manually specifying number of lines. Is there an easy way to do this in `yq` or would it also involve a nasty query string?

**Comment by mikefarah** [comment]
Without having a sample document it's hard to you the answer - it seems like you need to use the `select` operator (https://mikefarah.gitbook.io/yq/operators/select) and do something like:

```
yq 'select(.this.path.exists)' document.yaml
```

**Reply by chipzoller** [reply]
Thank you, Mike!
